src/charm.py

```diff
@@ -263,148 +332,125 @@ def _update_workload(self, event):
-        env_vars["OPENFGA_DATASTORE_ENGINE"] = "postgres"
-        env_vars["OPENFGA_DATASTORE_URI"] = self._state.db_uri
-
-        token = self._get_token(event)
-        if token:
-            env_vars["OPENFGA_AUTHN_METHOD"] = "preshared"
-            env_vars["OPENFGA_AUTHN_PRESHARED_KEYS"] = token
-
-        if self._state.certificate and self._state.private_key:
-            container.push("/app/certificate.pem", self._state.certificate, make_dirs=True)
-            container.push("/app/key.pem", self._state.private_key, make_dirs=True)
-            env_vars["OPENFGA_HTTP_TLS_ENABLED"] = "true"
-            env_vars["OPENFGA_HTTP_TLS_CERT"] = "/app/certificate.pem"
-            env_vars["OPENFGA_HTTP_TLS_KEY"] = "/app/key.pem"
-            env_vars["OPENFGA_GRPC_TLS_ENABLED"] = "true"
-            env_vars["OPENFGA_GRPC_TLS_CERT"] = "/app/certificate.pem"
-            env_vars["OPENFGA_GRPC_TLS_KEY"] = "/app/key.pem"
-
-        env_vars = {key: value for key, value in env_vars.items() if value}
-        for setting in REQUIRED_SETTINGS:
-            if not env_vars.get(setting, ""):
-                self.unit.status = BlockedStatus(
-                    "{} configuration value not set".format(setting),
+                openfga_relation.data[self.app].update(
+                    {
+                        "address": self._get_address(openfga_relation),
+                        "dns-name": self._domain_name,
+                    }
                 )
-                return
 
-        pebble_layer = {
-            "summary": "openfga layer",
-            "description": "pebble layer for openfga",
-            "services": {
-                SERVICE_NAME: {
-                    "override": "merge",
-                    "summary": "OpenFGA",
-                    "command": "sh -c 'openfga run | tee {LOG_FILE}'",
-                    "startup": "disabled",
-                    "environment": env_vars,
-                }
-            },
-            "checks": {
-                "openfga-check": {
-                    "override": "replace",
-                    "period": "1m",
-                    "http": {"url": "http://localhost:8080/healthz"},
-                }
-            },
-        }
-        container.add_layer("openfga", pebble_layer, combine=True)
-        if self._ready():
-            if container.get_service(SERVICE_NAME).is_running():
-                container.replan()
-            else:
-                container.start(SERVICE_NAME)
-            self.unit.status = ActiveStatus()
-            if self.unit.is_leader():
-                self.app.status = ActiveStatus()
-        else:
+        self._container.add_layer("openfga", self._pebble_layer, combine=True)
+        if not self._ready():
             logger.info("workload container not ready - deferring")
             event.defer()
             return
 
-    def _on_peer_relation_changed(self, event):
+        self._container.restart(SERVICE_NAME)
```

Comment:
I would put it in a try except block, in case it fails to restart
```suggestion
        try:
            self._container.restart(SERVICE_NAME)
        except ChangeError as err:
            logger.error(str(err))
            self.unit.status = BlockedStatus("Failed to restart the container, please consult the logs")
            return
```